cmd/minikube/cmd/start.go

```diff
@@ -1244,6 +1248,20 @@ func validateListenAddress(listenAddr string) {
 	}
 }
 
+// This function validates if the --listen-apiserver-port
+// is valid
+func validateListenAPIServerPort(listenAPIServerPort int) {
+	if listenAPIServerPort < 0 || listenAPIServerPort > 65535 {
+		exit.Message(reason.Usage, "Sorry, the port provided with the --listen-apiserver-port flag is invalid: {{.listenAPIServerPort}}.", out.V{"listenAPIServerPort": listenAPIServerPort})
+	} else {
+		ln, err := net.Listen("tcp", fmt.Sprintf(":%d", listenAPIServerPort))
+		if err != nil {
+			exit.Message(reason.Usage, "Sorry, the port provided with the --listen-apiserver-port flag is already allocated: {{.listenAPIServerPort}}.", out.V{"listenAPIServerPort": listenAPIServerPort})
+		}
+		defer ln.Close()
```

Comment:
this defer needs to happen before Exit

Reply:
Hi @medyagh 
Thanks so much for your time.

If it goes into `exit.Message`, that means the socket wouldn't be created. Do we still need to modify this snippet?
You're far more familiar to Golang that me. I referred the logic in [this file, example_test.go](https://golang.org/src/net/example_test.go), at line 23.

Besides, I noticed #9404 has achieved this feature. The only drawback is that #9404 exposes one more random port.
Should we still work on this feature, `--listen-apiserver-port`?

Please let me know what you think.

Reply:
@zhan9san You are correct about `defer`, if an error is returned no action is required on the socket.